javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,136 @@
+/*
+ * Javascript file for Hack Your Future, Javascript1, homework Week4
+ * program will return a response that corresponds to the command.
+*/
+
+// Voice assistant
+
+let userName = "";
+let toDo = [];
+
+function getReply(command) {
+
+    if (typeof command !== "string") {
+        return "Please retype your Question";
+    }
+    let str = command.split(" ");
+    if (command.includes("Hello my name is ")) {
+
+        if (userName === str[str.length - 1]) {
+            return `Hi again. You have registered before.`
+        } else {
+            userName = str[str.length - 1];
+            return `nice to meet you ${userName}.`;
+        }
+
+    } else if (command === "What is my name") {
+        if (!userName) {
+            return `Your have not registered your name yet.`;
+
+        } else {
+            return `Your name is ${userName}.`;
+        }
+
+
+    }
+    if (command.includes("to my todo") || command.includes("from my todo")) {
+        let keyWord = str[0];
+        let activity;
+        let commandIndex;
+
+        switch (keyWord) {
+            case "Add":
+                commandIndex = command.indexOf("to my todo");
+                activity = command.slice(4, commandIndex);
+                toDo.push(activity);
+                return `${activity}added to your todo`;
+
+                break;
+
+            case "Remove":
+                commandIndex = command.indexOf("from my todo");
+                activity = command.slice(7, commandIndex);
+                toDo.splice(toDo.indexOf(activity), 1);
+                return `Removed ${activity} from your todo`;
+
+                break;
+        }
+    }
+    if (command == "What is on my todo?") {
+        return `you have ${toDo.length} todos - ${toDo.join("and ")}`;
+    }
+    if (command === "What day is it today?") {
+        const today = new Date();
+        const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+        let month = months[today.getMonth()];
+
+        return `${today.getDate()}. ${month} ${today.getFullYear()}`;
+
+    }
+
+    let math = str[str.length - 2];
+    let a = parseInt(str[str.length - 3]);
+    let b = parseInt(str[str.length - 1]);
+    switch (math) {
+        case "+": return a + b;
+
+            break;
+
+        case "-": return a - b;
+
+            break;
+
+        case "*": return a * b;
+
+            break;
+
+        case "/": return a / b;
+
+            break;
+
+    }
```

Comment:
What if `math` is an unknown operator like `math = $`? Add a `default` case to the switch statement to handle all math operators that are not know. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch